Neos.Neos/Classes/Security/Authorization/ContentRepositoryAuthorizationService.php

```diff
@@ -34,133 +33,71 @@
+        $subjects = array_map(WorkspaceRoleSubject::createForGroup(...), $roleIdentifiers);
+        if ($userId !== null) {
+            $subjects[] = WorkspaceRoleSubject::createForUser($userId);
         }
+        $userIsAdministrator = array_key_exists(self::ROLE_NEOS_ADMINISTRATOR, $roleIdentifiers);
+
         $userWorkspaceRole = $this->workspaceService->getMostPrivilegedWorkspaceRoleForSubjects($contentRepositoryId, $workspaceName, WorkspaceRoleSubjects::fromArray($subjects));
         if ($userWorkspaceRole === null) {
             if ($userIsAdministrator) {
-                return WorkspacePermissions::manage(sprintf('Account "%s" is a Neos Administrator without explicit role for workspace "%s"', $account->getAccountIdentifier(), $workspaceName->value));
+                return WorkspacePermissions::manage(sprintf('User is a Neos Administrator without explicit role for workspace "%s"', $workspaceName->value));
             }
-            return WorkspacePermissions::none(sprintf('Account "%s" is no Neos Administrator and has no explicit role for workspace "%s"', $account->getAccountIdentifier(), $workspaceName->value));
+            return WorkspacePermissions::none(sprintf('User is no Neos Administrator and has no explicit role for workspace "%s"', $workspaceName->value));
         }
         return WorkspacePermissions::create(
             read: $userWorkspaceRole->isAtLeast(WorkspaceRole::VIEWER),
             write: $userWorkspaceRole->isAtLeast(WorkspaceRole::COLLABORATOR),
             manage: $userIsAdministrator || $userWorkspaceRole->isAtLeast(WorkspaceRole::MANAGER),
-            reason: sprintf('Account "%s" is %s Neos Administrator and has role "%s" for workspace "%s"', $account->getAccountIdentifier(), $userIsAdministrator ? 'a' : 'no', $userWorkspaceRole->value, $workspaceName->value),
+            reason: sprintf('User is %s Neos Administrator and has role "%s" for workspace "%s"', $userIsAdministrator ? 'a' : 'no', $userWorkspaceRole->value, $workspaceName->value),
         );
     }
 
-    public function getNodePermissionsForAnonymousUser(Node $node): NodePermissions
-    {
-        $roles = $this->rolesOfAnonymousUser();
-        return $this->nodePermissionsForRoles($node, $roles);
-    }
-
-    public function getNodePermissionsForAccount(Node $node, Account $account): NodePermissions
-    {
-        $roles = $this->expandAccountRoles($account);
-        return $this->nodePermissionsForRoles($node, $roles);
-    }
-
     /**
-     * Determines the default {@see VisibilityConstraints} for an anonymous user (aka "public access")
+     * Determines the {@see NodePermissions} a user with the specified {@see Role}s has on the given {@see Node}
+     *
+     * @param array<Role> $roles
      */
-    public function getVisibilityConstraintsForAnonymousUser(ContentRepositoryId $contentRepositoryId): VisibilityConstraints
+    public function getNodePermissions(Node $node, array $roles): NodePermissions
```

Comment:
A sorry my bad `NodePermissions` should be `@api` after all if its returned from this `@api` service? And so `WorkspacePermissions`?

Reply:
True, adjusted with 6397ff4ec27caa004cc796122cea44e7bbb8c762